### PR TITLE
libNX_X11: upgrade to X.org upstream version 1.6.6

### PR DESCRIPTION
**Pull request by uli42**
We are at X.Org libX11 upstream commit 733f64b

Fixes: ArcticaProject/nx-libs #716, 719 and #720

### CODE REVIEW

File: nx-X11/lib/src/XlibInt.c [hunk 1]
```diff
@@ -2083,30 +2087,31 @@ _XSetLastRequestRead(
...
+	if (newseq > X_DPY_GET_REQUEST(dpy)) {
 #ifdef NX_TRANS_SOCKET
 	    if (_NXLostSequenceFunction != NULL)
-            {
-                (*_NXLostSequenceFunction)(dpy, newseq, dpy->request,
-                                               (unsigned int) rep->type);
-            }
-            else
+	    {
+		(*_NXLostSequenceFunction)(dpy, newseq, dpy->request,
```
Comment (sunweaver) [inline nx-X11/lib/src/XlibInt.c]: Shouldn't we use X_DPY_GET_REQUEST here?

File: nx-X11/lib/src/XlibInt.c [hunk 2]
```diff
@@ -2921,15 +2926,12 @@ void _XEnq(
...
 		dpy->qfree = qelt->next;
 	}
-        else if ((qelt = Xmalloc(sizeof(_XQEvent))) == NULL) {
+	else if ((qelt = Xmalloc(sizeof(_XQEvent))) == NULL) {
 		/* Malloc call failed! */
 		ESET(ENOMEM);
-#ifdef NX_TRANS_SOCKET
                 _XIOError(dpy);
-
+#ifdef NX_TRANS_SOCKET
```
Comment (sunweaver) [inline nx-X11/lib/src/XlibInt.c:2933]: This doesn't look correct here... check the ifdef NX_TRANS_SOCKET block once more.

**Review by sunweaver — CHANGES_REQUESTED** [review]
Two minor comments. Please revisit.

**Comment by sunweaver** [comment]
Please close #716 and #719 with the commit in this PR. Thanks

File: nx-X11/lib/src/XlibInt.c (hunk 1 above)
Comment (uli42) [reply nx-X11/lib/src/XlibInt.c]: yes, I think so!

File: nx-X11/lib/src/XlibInt.c (hunk 2 above)
Comment (uli42) [reply nx-X11/lib/src/XlibInt.c:2933]: Before:
```
#ifdef NX_TRANS_SOCKET
                _XIOError(dpy);
                return;
#else
                _XIOError(dpy);
#endif
```

Now:
```
                 _XIOError(dpy);
+#ifdef NX_TRANS_SOCKET
                 return;
 #endif
```

I'd call that equivalent, right?

**Comment by uli42** [comment]
What do you mean? The commit message contains #716, #719 and #720

**Comment by sunweaver** [comment]
> What do you mean? The commit message contains #716, #719 and #720

You are right. Only the PR teaser text mentioned #720 only. So, all good. Please update the X_DPY_GET_REQUEST passage and we are good to go for a merge.

**Comment by sunweaver** [comment]
Hi,

On  Mo 22 Okt 2018 15:29:29 CEST, Ulrich Sibiller wrote:

> uli42 commented on this pull request.
>
>
>
>>                  _XIOError(dpy);
> -
> +#ifdef NX_TRANS_SOCKET
>
> Before:
> ```
> #ifdef NX_TRANS_SOCKET
>                 _XIOError(dpy);
>                 return;
> #else
>                 _XIOError(dpy);
> #endif
> ```
>
> Now:
> ```
>                  _XIOError(dpy);
> +#ifdef NX_TRANS_SOCKET
>                  return;
>  #endif
> ```
>
> I'd call that equivalent, right?

Yes, my brain got twisted on that one...

Ack.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

File: nx-X11/lib/src/XlibInt.c (hunk 1 above)
Comment (uli42) [reply nx-X11/lib/src/XlibInt.c]: done

**Comment by uli42** [comment]
ok, requested change is done

**Comment by uli42** [comment]
Turned out there's only one commit missing to 1.6.7 so I added that one, too.

File: nx-X11/lib/src/XlibInt.c (hunk 2 above)
Comment (sunweaver) [reply nx-X11/lib/src/XlibInt.c:2933]: Ok. You are right.

**Review by sunweaver — APPROVED** [review]
Looks good now. Merging.

**Pull request merged**